sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -178,6 +179,27 @@ class Analyzer(
     }
   }
 
+  /**
+   * Replaces queries of the form "SELECT expr FROM A GROUP BY 1"
+   * with a query of the form "SELECT expr FROM A GROUP BY expr"
+   */
+  object ResolveGroupByReferences extends Rule[LogicalPlan] {
+
+    def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
+      case Aggregate(groups, aggs, child) =>
+        Aggregate(groups.map(group => group match {
+          case g if g.prettyString forall Character.isDigit =>
```

Comment:
I'm not sure we want to match on the pretty string as I think this would also trigger for things like `"1"`, instead I'd consider matching on a Literal of NumericType.